qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -0,0 +1,391 @@
+            or "resistance". All parameters and methods in this submodule
+            should only be accessible to the user if
+            self.parent.sense_function.get() == self._proper_function. We
+            ensure this through the 'sense' property on the main driver class
+            which returns the proper submodule for any given function mode
+    """
+
+    function_modes = {
+        "current": {
+            "name": '"CURR:DC"',
+            "unit": "A",
+            "range_vals": Numbers(10E-9, 1)
+        },
+        "resistance": {
+            "name": '"RES"',
+            "unit": "Ohm",
+            "range_vals": Numbers(20, 200E6)
+        },
+        "voltage": {
+            "name": '"VOLT:DC"',
+            "unit": "V",
+            "range_vals": Numbers(0.02, 200)
+        }
+    }
+
+    def __init__(self, parent: 'Keithley2450', name: str, proper_function: str) -> None:
+        super().__init__(parent, name)
+        self._proper_function = proper_function
+        range_vals = self.function_modes[self._proper_function]["range_vals"]
+        unit = self.function_modes[self._proper_function]["unit"]
+
+        self.function = self.parent.sense_function
+
+        self.add_parameter(
+            "four_wire_measurement",
+            set_cmd=f":SENSe:{self._proper_function}:RSENse {{}}",
+            get_cmd=f":SENSe:{self._proper_function}:RSENse?",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0")
+        )
+
+        self.add_parameter(
+            "range",
+            set_cmd=f":SENSe:{self._proper_function}:RANGe {{}}",
+            get_cmd=f":SENSe:{self._proper_function}:RANGe?",
+            vals=range_vals,
+            get_parser=float,
+            unit=unit
+        )
+
+        self.add_parameter(
+            "auto_range",
+            set_cmd=f":SENSe:{self._proper_function}:RANGe:AUTO {{}}",
+            get_cmd=f":SENSe:{self._proper_function}:RANGe:AUTO?",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0")
+        )
+
+        self.add_parameter(
+            self._proper_function,
+            get_cmd=self._measure,
+            get_parser=float,
+            unit=unit,
+            snapshot_value=False
+        )
+
+        self.add_parameter(
+            "sweep",
+            label=self._proper_function,
+            get_cmd=self._measure_sweep,
+            unit=unit,
+            vals=Arrays(shape=(self.parent.npts,)),
+            parameter_class=ParameterWithSetpoints
+        )
+
+    def _measure(self) -> Callable:
+        if not self.parent.output_enabled():
+            raise RuntimeError("Output needs to be on for a measurement")
+        return self.ask(":MEASure?")
+
+    def _measure_sweep(self) -> np.ndarray:
+
+        source = cast(Source2450, self.parent.source)
+        source.sweep_start()
+        raw_data = self.ask(f":TRACe:DATA? 1, {self.parent.npts()}")
+        self.write(":TRACe:CLEar")
```

Comment:
what is this command needed for? why is it necessary to clear the trace RIGHT AFTER the measurement? is it ok that this happens all the time, or shall this be made optional?

Reply:
I want to avoid old data lingering around and potentially contaminating a subsequent measurement. We want to minimize unwanted statedness.

Reply:
i understand that. I just wonder when and how to that in the right way.

ok, if so far this hasn't given problems, let's keep it as is, but please make it a separate method of the driver (for explicitness).